packages/core/src/new-api/stored-deployment-serializer.ts

```diff
@@ -80,21 +91,16 @@ export class StoredDeploymentSerializer {
   ): SerializedStoredModule {
     return {
       id: userModule.id,
-      futures: Object.fromEntries(
-        Array.from(userModule.futures).map((future) => [
-          future.id,
-          this._serializeFuture(future),
-        ])
+      futures: Array.from(userModule.futures).map((future) =>
```

Comment:
I modified the serialization to preserver the other of:

1. Module's futures
2. Module's submodules
3. Module's results
4. Future's libraries

This allowed me to make the tests stricter with `deepEqual`. They failed otherwise, as sets and objects have an order in javascript.